client/app/src/main/java/com/microsoft/research/karya/ui/scenarios/speechData/SpeechDataMainFragment.kt

```diff
@@ -0,0 +1,296 @@
+    }
+
+    playBtnCv.addOnLayoutChangeListener { _: View, left: Int, _: Int, right: Int, _: Int, _: Int, _: Int, _: Int, _: Int
+      ->
+      playBtnCv.radius = (right - left).toFloat() / 2
+    }
+
+    /** Set on click listeners */
+    recordBtn.setOnClickListener { viewModel.handleRecordClick() }
+    playBtn.setOnClickListener { viewModel.handlePlayClick() }
+    nextBtn.setOnClickListener { viewModel.handleNextClick() }
+    backBtn.setOnClickListener { viewModel.handleBackClick() }
+
+  }
+
+  private fun setupObservers() {
+    viewModel.backBtnState.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { state ->
+      backBtn.isClickable = state != DISABLED
+      backBtn.setBackgroundResource(
+        when (state) {
+          DISABLED -> R.drawable.ic_back_disabled
+          ENABLED -> R.drawable.ic_back_enabled
+          ACTIVE -> R.drawable.ic_back_enabled
+        }
+      )
+    }
+
+    viewModel.recordBtnState.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { state ->
+      recordBtn.isClickable = state != DISABLED
+      recordBtn.setBackgroundResource(
+        when (state) {
+          DISABLED -> R.drawable.ic_mic_disabled
+          ENABLED -> R.drawable.ic_mic_enabled
+          ACTIVE -> R.drawable.ic_mic_enabled
+        }
+      )
+    }
+
+    viewModel.playBtnState.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { state ->
+      playBtn.isClickable = state != DISABLED
+      playBtn.setBackgroundResource(
+        when (state) {
+          DISABLED -> R.drawable.ic_speaker_disabled
+          ENABLED -> R.drawable.ic_speaker_enabled
+          ACTIVE -> R.drawable.ic_speaker_enabled
+        }
+      )
+    }
+
+    viewModel.nextBtnState.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { state ->
+      nextBtn.isClickable = state != DISABLED
+      nextBtn.setBackgroundResource(
+        when (state) {
+          DISABLED -> R.drawable.ic_next_disabled
+          ENABLED -> R.drawable.ic_next_enabled
+          ACTIVE -> R.drawable.ic_next_enabled
+        }
+      )
+    }
+
+    viewModel.sentenceTvText.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { text ->
+      sentenceTv.text = text
+    }
+
+    viewModel.recordSecondsTvText.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { text ->
+      recordSecondsTv.text = text
+    }
+
+    viewModel.recordCentiSecondsTvText.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { text ->
+      recordCentiSecondsTv.text = text
+    }
+
+    viewModel.playbackProgressPb.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { progress ->
+      playbackProgressPb.progress = progress
+    }
+
+    viewModel.playbackProgressPbMax.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { max ->
+      playbackProgressPb.max = max
+    }
+
+    viewModel.playRecordPromptTrigger.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { play ->
+      if (play) {
+        playRecordPrompt()
+      }
+    }
+
```

Comment:
I belive it's better to pass all this in a single data class rather than passing it individually using multiple flows.

Also, all of these are using incorrect `lifecycleScope` they should use `viewLifecycle` and `viewLifecycleScope`